Tests/Tests/Encryption/Keychain/CDTEncryptionKeychainDataTests.m

```diff
@@ -0,0 +1,133 @@
+    self.defaultSalt = nil;
+    self.defaultIv = nil;
+    self.defaultVersion = nil;
+
+    [super tearDown];
+}
+
+- (void)testSimpleInitFails
+{
+    XCTAssertNil([[CDTEncryptionKeychainData alloc] init], @"All properties are mandatory");
+}
+
+- (void)testInitFailsIfEncryptedDPKIsNil
+{
+    XCTAssertNil([[CDTEncryptionKeychainData alloc] initWithEncryptedDPK:nil
+                                                                    salt:self.defaultSalt
+                                                                      iv:self.defaultIv
+                                                              iterations:self.defaultIterations
+                                                                 version:self.defaultVersion],
+                 @"Encrypted DPK is mandatory");
+}
+
+- (void)testInitFailsIfSaltIsNil
+{
+    XCTAssertNil([[CDTEncryptionKeychainData alloc] initWithEncryptedDPK:self.defaultEncryptedDPK
+                                                                    salt:nil
+                                                                      iv:self.defaultIv
+                                                              iterations:self.defaultIterations
+                                                                 version:self.defaultVersion],
+                 @"Salt is mandatory");
+}
+
+- (void)testInitFailsIfIVIsNil
+{
+    XCTAssertNil([[CDTEncryptionKeychainData alloc] initWithEncryptedDPK:self.defaultEncryptedDPK
+                                                                    salt:self.defaultSalt
+                                                                      iv:nil
+                                                              iterations:self.defaultIterations
+                                                                 version:self.defaultVersion],
+                 @"IV is mandatory");
+}
+
+- (void)testInitFailsIfVersionIsNil
+{
+    XCTAssertNil([[CDTEncryptionKeychainData alloc] initWithEncryptedDPK:self.defaultEncryptedDPK
+                                                                    salt:self.defaultSalt
+                                                                      iv:self.defaultIv
+                                                              iterations:self.defaultIterations
+                                                                 version:nil],
+                 @"IV is mandatory");
+}
+
+- (void)testInitFailsIfIterationsValueIsNegative
+{
+    XCTAssertNil([[CDTEncryptionKeychainData alloc] initWithEncryptedDPK:self.defaultEncryptedDPK
+                                                                    salt:self.defaultSalt
+                                                                      iv:self.defaultIv
+                                                              iterations:-1
+                                                                 version:self.defaultVersion],
+                 @"Iterations value has to be a positive number");
```

Comment:
Probably for the error message too, but this should be "Iterations value has to be positive or zero"